### ISSUE TITLE
Mocking the LinkProvider

### ISSUE THREAD

**Issue by sergeyshushlyapin** [sergeyshushlyapin/Sitecore.FakeDb#89]
The default `Sitecore.Links.LinkProvider` has been in use from the beginning of the FakeDb project (thanks to @pveller ;)). As soon as one has a `db` context instantiated, the LinkManager can be used to get an item URLs. That is OK for most of the cases when *some* url needs to be returned.

From the other side, the logic **inside** the provider is very complex. The final URL depends from numerous factors such as configuration in the `App.config` file, current HttpContext, Context Site etc.

Sometimes in **isolated** unit tests it's enough to check that the LinkManager has been called with proper arguments. Using a mocked provider looks like the easiest option in this case:
```csharp
using (var db = new Db { new DbItem("/sitecore/content/home") })
{
  var item = db.GetItem("/sitecore/content/home");

  var provider = Substitute.For<LinkProvider>();

  // configure the provider mock
  provider.GetItemUrl(item,
    Arg.Is<UrlOptions>(x => x.AlwaysIncludeServerUrl)).Returns("http://myawesomeurl.com");

  using (new LinkProviderSwitcher(provider))
  {
    var options = new UrlOptions { AlwaysIncludeServerUrl = true };
    var result = LinkManager.GetItemUrl(item, options);

    Assert.Equal("http://myawesomeurl.com", result);
  }
}
```

**Comment by sergeyshushlyapin** [comment]
And there is a code snippet for funs of AutoFixture:
```csharp
[Theory, DefaultAutoData]
public void SutSwitchesToMockedLinkProvider([Substitute]LinkProvider provider,
  Item item, [NoAutoProperties]UrlOptions options)
{
  provider.GetItemUrl(item, options).Returns("http://myawesomeurl.com");

  using (new LinkProviderSwitcher(provider))
  {
    LinkManager.GetItemUrl(item, options).Should().Be("http://myawesomeurl.com");
  }
}
```

**Comment by sergeyshushlyapin** [comment]
Done.

**Comment by sergeyshushlyapin** [comment]
[Documented](https://github.com/sergeyshushlyapin/Sitecore.FakeDb/wiki/Links#mocking-the-linkprovider) and released as [v0.31.0](https://github.com/sergeyshushlyapin/Sitecore.FakeDb/releases/tag/v0.31.0)
